xpath.d.ts

```diff
@@ -1,11 +1,35 @@
 /// <reference lib="dom" />
 
-type SelectedValue = Node | Attr | string | number | boolean;
-interface XPathSelect {
-    (expression: string, node?: Node): Array<SelectedValue>;
-    (expression: string, node: Node, single: true): SelectedValue | undefined;
+export type SelectedValue = Node | string | number | boolean | null;
+
+export interface XPathSelect {
+    (expression: string, node: Node): Array<Node>;
+    (expressed: string, node: Node, single: false): Array<Node>;
+    (expression: string, node: Node, single: true): SelectedValue;
 }
-export var select: XPathSelect;
-export function select1(expression: string, node?: Node): SelectedValue | undefined;
-export function evaluate(expression: string, contextNode: Node, resolver: XPathNSResolver | null, type: number, result: XPathResult | null): XPathResult;
-export function useNamespaces(namespaceMap: { [name: string]: string }): XPathSelect;
+
+export type SelectWithResolverReturnType = Array<Node> | SelectedValue;
+export type SelectWithResolverSingleReturnType = SelectedValue;
+
+export function select(expression: string, node: Node): SelectWithResolverReturnType;
+export function select(expression: string, node: Node, single: false): SelectWithResolverReturnType;
+export function select(expression: string, node: Node, single: true): SelectWithResolverSingleReturnType;
+
+export function select1(expression: string, node: Node): SelectWithResolverSingleReturnType;
+
+export function useNamespaces(namespaceMap: Record<string, string>): XPathSelect;
+
+export function selectWithResolver(expression: string, node: Node, resolver?: XPathNSResolver | null): SelectWithResolverReturnType;
```

Comment:
Seems like it would make sense to include an interface definition for `XPathNSResolver` in this file.

I think it would be 

```ts
interface XPathNSResolver {
    lookupNamespaceURI(prefix: string): string;
}
```

Reply:
Technically, the build-in DOM types already include this. As long as you're using something compatible, we can just use the built-in types.

From `lib.dom.d.ts`
```
type XPathNSResolver = ((prefix: string | null) => string | null) | { lookupNamespaceURI(prefix: string | null): string | null; };
```